src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java

```diff
@@ -62,73 +64,123 @@ public Boolean getTspEligible() {
 
         List<AssessmentNeed> assessmentNeeds = new ArrayList<>();
 
         for (Section section : filteredSections) {
-            Map<String, String> planSection = planSections.get(section.getRefSectionCode());
-            boolean riskHarm = Optional.ofNullable(section.getQuestions().get(planSection.get("harmQuestion"))).map(e -> e.getAnswer().get()).map(e -> e.getRefAnswerCode()).orElse("NO").equals("YES");
-            boolean riskReoffending =  Optional.ofNullable(section.getQuestions().get(planSection.get("reoffendingQuestion"))).map(e -> e.getAnswer().get()).map(e -> e.getRefAnswerCode()).orElse("NO").equals("YES");
-            boolean overThreshold = Optional.ofNullable(section.getSectionOtherRawScore()).orElse(0L) >= Optional.ofNullable(section.getRefSection().getRefCrimNeedScoreThreshold()).orElse(Long.MAX_VALUE);
-            boolean flagged = Optional.ofNullable(section.getLowScoreAttentionNeeded()).orElse(false);
+            Map<String, String> planSection = Optional.ofNullable(planSections.get(section.getRefSectionCode()))
+                    .orElse(new HashMap<>());
+
+            var riskHarm = sectionHasRiskOfHarm(section, planSection);
+            var riskReoffending = sectionHasRiskOfReoffending(section, planSection);
+            var overThreshold = sectionIsOverThreshold(section);
+            var flagged = Optional.ofNullable(section.getLowScoreAttentionNeeded()).orElse(false);
+
             if (Stream.of(overThreshold, riskHarm, riskReoffending, flagged).anyMatch(e -> e.equals(true))) {
-                assessmentNeeds.add(new AssessmentNeed(section.getRefSection().getShortDescription(), overThreshold, riskHarm, riskReoffending, flagged));
+                assessmentNeeds.add(new AssessmentNeed(section.getRefSection().getShortDescription(),
+                        overThreshold, riskHarm, riskReoffending, flagged));
             }
+
+            questionHarmNeeds.entrySet().stream().filter(e -> e.getValue().equals(section.getRefSectionCode())).forEach(e -> {
+                Optional<Question> question = Optional.ofNullable(section.getQuestions().get(e.getKey()));
+                if (questionHasAnswer(question, "YES")) {
+                    assessmentNeeds.add(new AssessmentNeed(question.map(q -> q.getQuestionText()).orElse(""), false, true, false, false));
+                }
+            });
+
         }
         return assessmentNeeds;
     }
 
+
+    private boolean sectionIsOverThreshold(Section section) {
+        return Optional.ofNullable(section.getSectionOtherRawScore()).orElse(0L) >= Optional.ofNullable(section.getRefSection()).map(s->s.getRefCrimNeedScoreThreshold()).orElse(Long.MAX_VALUE);
+    }
+
+    private boolean sectionHasRiskOfReoffending(Section section, Map<String, String> planSection) {
+        return questionHasAnswer(Optional.ofNullable(section.getQuestions().get(Optional.ofNullable(planSection.get("reoffendingQuestion")).orElse(""))),"YES");
```

Comment:
and this, etc?

Reply:
The answers in OASys are always YES or NO for these sections so safe to default to empty string when no answer is present as that will then set the risk flag to false by default. 

The key thing here is that for the assessment to be complete there will always be an answer except where there is bad data in OASys, the defaults are just guarding against that scenario. The alternative is to error and not return an assessment but an error response.